protocol-designer/src/components/StepEditForm/fields/TiprackField.tsx

```diff
@@ -1,32 +1,62 @@
 import * as React from 'react'
 import { useTranslation } from 'react-i18next'
 import { useSelector } from 'react-redux'
-import { FormGroup, DropdownField } from '@opentrons/components'
+import {
+  FormGroup,
+  DropdownField,
+  useHoverTooltip,
+  Tooltip,
+  Box,
+} from '@opentrons/components'
 import { selectors as uiLabwareSelectors } from '../../../ui/labware'
+import { getPipetteEntities } from '../../../step-forms/selectors'
 import styles from '../StepEditForm.module.css'
 
 import type { FieldProps } from '../types'
 
-export function TiprackField(props: FieldProps): JSX.Element {
-  const { name, value, onFieldBlur, onFieldFocus, updateValue } = props
-  const { t } = useTranslation('form')
+interface TiprackFieldProps extends FieldProps {
+  pipetteId?: unknown
+}
+export function TiprackField(props: TiprackFieldProps): JSX.Element {
+  const {
+    name,
+    value,
+    onFieldBlur,
+    onFieldFocus,
+    updateValue,
+    pipetteId,
+  } = props
+  const { t } = useTranslation(['form', 'tooltip'])
+  const [targetProps, tooltipProps] = useHoverTooltip()
+  const pipetteEntities = useSelector(getPipetteEntities)
   const options = useSelector(uiLabwareSelectors.getTiprackOptions)
+  const defaultTipracks =
+    pipetteId != null ? pipetteEntities[pipetteId as string].tiprackDefURI : []
+  const pipetteOptions = options.filter(option =>
+    defaultTipracks.includes(option.defURI)
+  )
+  const missingTiprack = defaultTipracks.length > pipetteOptions.length
```

Comment:
nit 
maybe adding `is` or `has`?

```suggestion
  const missingTiprack = defaultTipracks.length > pipetteOptions.length
```